File: problemas/1000/1049.py
palavra = input()

if palavra == 'vertebrado':
    palavra = input()
    if palavra == 'ave':
        palavra = input()
        animal = 'aguia' if palavra == 'carnivoro' else 'pomba'
    else:
        palavra = input()
        animal = 'homem' if palavra == 'onivoro' else 'vaca'
else:
    palavra = input()
    if palavra == 'inseto':
        palavra = input()
        animal = 'pulga' if palavra == 'hematofago' else 'lagarta'
    else:
        palavra = input()
        animal = 'sanguessuga' if palavra == 'hematofago' else 'minhoca'

print(animal)

File: problemas/1000/1049.php
<?php

$palavra = readline();

if ($palavra == 'vertebrado')
{
    $palavra = readLine();
    if ($palavra == 'ave') {
        $palavra = readLine();
        $animal = $palavra == 'carnivoro' ? 'aguia' : 'pomba';
    } else {
        $palavra = readLine();
        $animal = $palavra == 'onivoro' ? 'homem' : 'vaca';
    }
} else {
    $palavra = readLine();
    if ($palavra == 'inseto') {
        $palavra = readLine();
        $animal = $palavra == 'hematofago' ? 'pulga' : 'lagarta';
    } else {
        $palavra = readLine();
        $animal = $palavra == 'hematofago' ? 'sanguessuga' : 'minhoca';
    }
}

echo ("{$animal}\n");